aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/MutationProcessor.java

```diff
@@ -316,7 +316,11 @@ void stopDrainingMutationOutbox() {
             @NonNull PublicationStrategy<T> publicationStrategy) {
         return Single
             .<GraphQLResponse<ModelWithMetadata<T>>>create(subscriber ->
-                publicationStrategy.publish(mutation.getMutatedItem(), subscriber::onSuccess, subscriber::onError)
+                publicationStrategy.publish(mutation.getMutatedItem(), subscriber::onSuccess, (exception) -> {
+                    if (!subscriber.isDisposed()) {
```

Comment:
Is there anywhere else in our pubsub model we may want to apply a similar strategy or checking for subscriber state? Or do all published events pass through this `publishWithStrategy` method?

Reply:
Good point, all the publish events pass through this method.